src/sequelize.js

```diff
@@ -628,36 +628,62 @@ class Sequelize {
 
     const retryOptions = { ...this.options.retry, ...options.retry };
 
-    return retry(async () => {
-      if (options.transaction === undefined && Sequelize._cls) {
-        options.transaction = Sequelize._cls.get('transaction');
-      }
+    let ret;
+    try {
+      ret = retry(async () => {
```

Comment:
nit: `ret` is only used inside this try-catch, it could be moved inside and returned at the end of `try`